app/controllers/programming_groups_controller.rb

```diff
@@ -65,21 +81,42 @@ def create
     end
   end
 
+  def update
+    myparams = programming_group_params2
+    myparams[:users] =
+      @programming_group.programming_group_memberships.where(id: myparams[:programming_group_membership_ids].compact_blank).includes(:user).map(&:user)
+    myparams.delete(:programming_group_membership_ids)
+    update_and_respond(object: @programming_group, params: myparams)
```

Comment:
In case the update is not successful, we need to set the members as well:

```rb
    @members = @programming_group.programming_group_memberships.includes(:user)
```

Reply:
For the StudyGroupsController#update, I fixed that as well. Potentially, you could update your code to be similar: https://github.com/openHPI/codeocean/commit/052367c1861028df5dcbe174da664175b7deee8f